### PR TITLE
Patch allow web payments

### PR DESCRIPTION
**Pull request by oswaldquek**

### CODE REVIEW

File: src/test/java/uk/gov/pay/connector/it/resources/GatewayAccountResourceITest.java [hunk 1]
```diff
@@ -570,4 +571,31 @@ public void shouldReturn400_whenNotifySettingsIsRemoved_withWrongPath() throws E
                 .then()
                 .statusCode(BAD_REQUEST.getStatusCode());
     }
+    
+    @Test
+    public void allow_web_payments_shouldBeFalseByDefault() {
+        String gatewayAccountId = createAGatewayAccountFor("worldpay", "old-desc", "old-id");
+        givenSetup()
```
Comment (simad) [inline src/test/java/uk/gov/pay/connector/it/resources/GatewayAccountResourceITest.java:577]: just fyi we also have another endpoint /v1/frontend/accounts etc. which is used in selfservice, and we might need to do additional work to surface the flag there. This is good enough for now.

File: src/main/java/uk/gov/pay/connector/resources/GatewayAccountRequestValidator.java [hunk 2]
```diff
@@ -53,6 +54,23 @@ void validatePatchRequest(JsonNode payload){
...
+        String op = payload.get(FIELD_OPERATION).asText();
+        if (!op.equalsIgnoreCase("replace")) {
+            throw new ValidationException(Collections.singletonList(format("Operation [%s] is not valid for path [%s]", op, FIELD_EMAIL_COLLECTION_MODE)));
+        }
+        JsonNode valueNode = payload.get(FIELD_VALUE);
+        if(null == valueNode || valueNode.isNull()) {
+            throw new ValidationException(Collections.singletonList(format("Field [%s] is required", FIELD_VALUE)));
+        }
+        String booleanString = valueNode.asText();
+        if (!booleanString.equalsIgnoreCase("false") && !booleanString.equalsIgnoreCase("true"))
```
Comment (simad) [inline src/main/java/uk/gov/pay/connector/resources/GatewayAccountRequestValidator.java]: this is a bit oof but #justjavathings

File: src/test/java/uk/gov/pay/connector/it/resources/GatewayAccountResourceITest.java (hunk 1 above)
Comment (simad) [reply src/test/java/uk/gov/pay/connector/it/resources/GatewayAccountResourceITest.java:577]: All you would need to do is add   @JsonProperty("allow_web_payments") to the getter, I think, and add a test in the frontend specific IT suite. :/

**Review by simad — DISMISSED** [review]
looks good

**Comment by oswaldquek** [comment]
I will address the comments so will merge later.

File: src/main/java/uk/gov/pay/connector/resources/GatewayAccountRequestValidator.java [hunk 3]
```diff
@@ -23,49 +22,42 @@
...
-            validateNotifySettingsRequest(payload);
-        }
-        if (path.equalsIgnoreCase(FIELD_EMAIL_COLLECTION_MODE)) {
-            validateEmailCollectionMode(payload);
+        
+        String path = payload.findValue(FIELD_OPERATION_PATH).asText().toLowerCase();
+        if(!VALID_PATHS.contains(path)) 
+            throw new ValidationException(Collections.singletonList(format("Operation [%s] not supported for path [%s]", FIELD_OPERATION, path)));
+        
+        switch (path) {
```
Comment (simad) [inline src/main/java/uk/gov/pay/connector/resources/GatewayAccountRequestValidator.java:48]: nice!

**Review by simad — COMMENTED** [review]
🚢 good work!

**Review by simad — APPROVED** [review]
😳 I can't even get this right

**Pull request merged**